src/configSchema.ts

```diff
@@ -0,0 +1,207 @@
+	const duration = ms(durationStr);
+	if (isNaN(duration)) {
+		// adds the error to the Zod Issues
+		zodAddIssue(durationStr, ZodErrorMessages.vercel, ctx);
+	}
+	return duration;
+}
+
+/**
+ * helper function for directory validation
+ * @return true if valid formatting
+ */
+function checkValidPathFormat(path: string, ctx) {
+	if (sep === "\\" && !path.includes("\\") && !path.includes("/")) {
+		zodAddIssue(path, ZodErrorMessages.windowsPath, ctx);
+	}
+	return path;
+}
+
+/**
+ * an object of the zod schema
+ * each are named after what they are intended to validate
+ */
+
+export const VALIDATION_SCHEMA = z
+	.object({
+		delay: z.number().positive({
+			message: ZodErrorMessages.delay,
+		}),
+		torznab: z.array(z.string().url()),
+		dataDirs: z
+			.array(
+				z
+					.string()
+					.transform((value, ctx) =>
+						value && value.length > 0
+							? checkValidPathFormat(value, ctx)
+							: null
+					)
+			)
+
+			.nullish(),
+		matchMode: z.nativeEnum(MatchMode),
+		dataCategory: z.string().nullish(),
+		linkDir: z.string().transform(checkValidPathFormat).nullish(),
+		linkType: z.nativeEnum(LinkType),
+		skipRecheck: z.boolean(),
+		maxDataDepth: z.number().gte(1),
+		torrentDir: z.string(),
+		outputDir: z.string(),
+		includeEpisodes: z.boolean(),
+		includeSingleEpisodes: z.boolean(),
+		includeNonVideos: z.boolean(),
+		fuzzySizeThreshold: z.number().positive().lte(1, {
+			message: ZodErrorMessages.fuzzySizeThreshold,
+		}),
+		excludeOlder: z
+			.string()
+			.min(1, { message: ZodErrorMessages.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		excludeRecentSearch: z
+			.string()
+			.min(1, { message: ZodErrorMessages.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		action: z.nativeEnum(Action),
+		qbittorrentUrl: z.string().url().nullish(),
+		rtorrentRpcUrl: z.string().url().nullish(),
+		transmissionRpcUrl: z.string().url().nullish(),
+		delugeRpcUrl: z.string().url().nullish(),
+		duplicateCategories: z.boolean(),
+		notificationWebhookUrl: z.string().url().nullish(),
+		port: z
+			.number()
+			.positive()
+			.lte(65535)
+			.or(z.boolean().transform(() => null))
+			.nullish(),
+		host: z.string().ip().nullish(),
+		rssCadence: z
+			.string()
+			.min(1, { message: ZodErrorMessages.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		searchCadence: z
+			.string()
+			.min(1, { message: ZodErrorMessages.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		snatchTimeout: z
+			.string()
+			.min(1, { message: ZodErrorMessages.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		searchTimeout: z
+			.string()
+			.min(1, { message: ZodErrorMessages.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		searchLimit: z.number().nonnegative().nullish(),
+		apiAuth: z.boolean().default(false),
+		verbose: z.boolean().default(false),
```

Comment:
```suggestion
		apiAuth: z.boolean().default(false),
		verbose: z.boolean().default(false),
```

Reply:
```
		apiAuth: z.boolean(),
		verbose: z.boolean(),
```

Reply:
u right, I deleted too much stuff

Reply:
Also if the value isn't in the zod object, it wont return it when we reset options to the validated object....so any config values we want to set in runtimeConfig in the future, they will HAVE to be set in Zod.

Just FYI.